channeldb/payment_control.go

```diff
@@ -296,6 +299,99 @@ func (p *PaymentControl) Success(paymentHash lntypes.Hash,
 	return payment, updateErr
 }
 
+// AttemptFailureReason is the reason a payment attemot failed.
+type AttemptFailureReason byte
+
+const (
+	// AttemptFailureReasonUnknown is recorded for attempts that failed
+	// with an unknown reason.
+	AttemptFailureReasonUnknown AttemptFailureReason = 0
+)
```

Comment:
Perhaps this should then be the htlc-level wire failure?

Reply:
Yeah, I think it is useful to keep track of the actual reason the HTLC failed. Maybe reuse the error codes added by @carlaKC ? :)

Reply:
Well, that is another thing again. I am just thinking about how we are going to return the raw failure in `SendToRoute`. If we decide to use `TrackPayment` to (reliably) track the outcome of a `SendToRoute` call (conceptually being more like`SendShard`), then we need that raw failure message in the db.